Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -246,4 +274,62 @@ ThrowCompletionOr<NonnullGCPtr<ArrayBuffer>> allocate_shared_array_buffer(VM& vm
     return obj;
 }
 
+// 1.1.3 ArrayBufferByteLength ( arrayBuffer, order ), https://tc39.es/proposal-resizablearraybuffer/#sec-arraybufferlength
+size_t array_buffer_byte_length(VM&, ArrayBuffer& array_buffer, ArrayBuffer::Order)
+{
+    // FIXME: 1. If IsSharedArrayBuffer(arrayBuffer) is true and arrayBuffer has an [[ArrayBufferByteLengthData]] internal slot, then
+    // a. Let bufferByteLengthBlock be arrayBuffer.[[ArrayBufferByteLengthData]].
+    // b. Return ℝ(GetValueFromBuffer(bufferByteLengthBlock, 0, BigUint64, true, order)).
+
+    // 2. Assert: IsDetachedBuffer(arrayBuffer) is false.
+    VERIFY(!array_buffer.is_detached());
+
+    // 3. Return arrayBuffer.[[ArrayBufferByteLength]].
+    return array_buffer.byte_length();
+}
+
+// 1.1.4 MakeIdempotentArrayBufferByteLengthGetter ( order ), https://tc39.es/proposal-resizablearraybuffer/#sec-makeidempotentarraybufferbytelengthgetter
+IdempotentArrayBufferByteLengthGetter make_idempotent_array_buffer_byte_length_getter(ArrayBuffer::Order order)
+{
+    // 1. NOTE: The [[ArrayBuffer]] slot is used for editorial clarity only, that a getter should only be used with a single ArrayBuffer.
+
+    // 2. Let lengthStorage be { [[ArrayBuffer]]: empty, [[ByteLength]]: empty }.
+    struct LengthStorage {
+        Optional<ArrayBuffer&> array_buffer;
+        Optional<size_t> byte_length;
+    };
+
+    auto length_storage = LengthStorage { {}, {} };
+
+    // 3. Let getter be a new Abstract Closure with parameters (buffer) that captures lengthStorage and order and performs the following steps when called:
+    auto getter = [length_storage, order](VM& vm, ArrayBuffer& buffer) mutable {
+        // a. If lengthStorage.[[ByteLength]] is empty, then
+        if (!length_storage.byte_length.has_value()) {
+            // i. Assert: lengthStorage.[[ArrayBuffer]] is empty.
+            VERIFY(!length_storage.array_buffer.has_value());
+
+            // ii. Set lengthStorage.[[ArrayBuffer]] to buffer.
+            length_storage.array_buffer = { buffer };
+
+            // iii. Set lengthStorage.[[ByteLength]] to ArrayBufferByteLength(buffer, order).
+            length_storage.byte_length = { array_buffer_byte_length(vm, buffer, order) };
+        }
+
+        // b. Assert: SameValue(lengthStorage.[[ArrayBuffer]], buffer) is true.
+        VERIFY(same_value(Value(&length_storage.array_buffer.value()), Value(&buffer)));
+
+        // c. Return lengthStorage.[[ByteLength]].
+        return length_storage.byte_length.value();
+    };
+
+    // 4. Return getter.
+    return getter;
+}
+
+// 1.1.7 HostResizeArrayBuffer ( buffer, newByteLength ) https://tc39.es/proposal-resizablearraybuffer/#sec-hostresizearraybuffer
+ThrowCompletionOr<void> host_resize_array_buffer(VM& vm, ArrayBuffer& array_buffer, size_t new_byte_length)
```

Comment:
Check how other `Host*` functions are implemented on the VM, this is not sufficient

Reply:
Done! host_resize_array_buffer is now defined in the VM in the same way other host functions are.